apps/qubit/modules/digitalobject/actions/showComponent.class.php

```diff
@@ -74,12 +74,19 @@ public function execute($request)
      */
     private function checkShowGenericIcon()
     {
+        $resourceObject = $this->resource->object;
+
+        // For digital objects, get the parent resouce (IO or Actor) for Acl check
+        if (!$resourceObject && ('QubitDigitalObject' === get_class($this->resource))) {
+            $resourceObject = $this->resource->parent->object;
+        }
```

Comment:
I think it has to be a `QubitDigitalObject` to be in this module as the resource, I'd change that to check that parent is not null, that way you kind of check the usage type of the DO.